AdyenComponents/Boleto/BoletoComponent.swift

```diff
@@ -0,0 +1,233 @@
+    private func getFormFields() -> [PersonalInformation] {
+        var fields: [PersonalInformation] = [
+            .custom(CustomFormItemInjector(item: headerFormItem(key: .boletoPersonalDetails))),
+            .firstName,
+            .lastName,
+            .custom(CustomFormItemInjector(item: socialSecurityNumberItem))
+        ]
+
+        if configuration.shopperInfo.billingAddress != nil {
+            fields.append(
+                .custom(CustomFormItemInjector(item: headerFormItem(key: .billingAddressSectionTitle)))
+            )
+            fields.append(.custom(CustomFormItemInjector(item: billingAddressLabelItem)))
+        } else {
+            fields.append(.address)
+        }
+        
+        if configuration.showEmailAddress {
+            fields.append( .custom(CustomFormItemInjector(item: sendCopyByEmailItem)))
+            fields.append(.email)
+        }
+        
+        return fields
+    }
+    
+    /// :nodoc:
+    /// Sets the initial values for the form fields based on configuration
+    private func setUpFields(for component: FormComponent) {
+        if let shopperName = configuration.shopperInfo.shopperName {
+            component.firstNameItem?.value = shopperName.firstName
+            component.lastNameItem?.value = shopperName.lastName
+        }
+        
+        if let socialSecurityNumber = configuration.shopperInfo.socialSecurityNumber {
+            socialSecurityNumberItem.value = socialSecurityNumber
+        }
+        
+        if let emailItem = component.emailItem {
+            sendCopyByEmailItem.value = false
+            emailItem.value = configuration.shopperInfo.emailAddress ?? ""
+            bind(sendCopyByEmailItem.publisher, to: emailItem, at: \.isHidden.wrappedValue, with: { !$0 })
+        }
+    }
+    
+    /// :nodoc:
+    private func createPaymentDetails() -> PaymentMethodDetails {
+        guard let firstNameItem = formComponent.firstNameItem,
+              let lastNameItem = formComponent.lastNameItem,
+              case let shopperName = ShopperName(firstName: firstNameItem.value, lastName: lastNameItem.value),
+              let billingAddress = configuration.shopperInfo.billingAddress ?? formComponent.addressItem?.value else {
+            fatalError("There seems to be an error in the BaseFormComponent configuration.")
+        }
+
+        return BoletoDetails(
+            type: paymentMethod.type,
+            shopperName: shopperName,
+            socialSecurityNumber: socialSecurityNumberItem.value,
+            emailAddress: getEmailDetails(),
+            billingAddress: billingAddress
+        )
+    }
+    
+    /// :nodoc:
+    /// Obtain email address depending if it was prefilled, or the checkbox was ticked
+    private func getEmailDetails() -> String? {
+        if let prefilledEmail = configuration.shopperInfo.emailAddress {
+            return prefilledEmail
+        } else if sendCopyByEmailItem.publisher.wrappedValue,
```

Comment:
you can use `sendCopyByEmailItem.value`, right?

Reply:
👍